src/components/UserFeed.jsx

```diff
@@ -1,69 +1,126 @@
+  const { token, logout, isLogged } = useAuth();
+  const [myUserFollowing, setMyUserFollowing] = useState([]);
   const [recipes, setRecipes] = useState([]);
+  const [FollowingRecipes, setFollowingRecipes] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [feedType, setFeedType] = useState("forYou");
+  const [maxRecipesPerUser, setMaxRecipesperUser] = useState(2);
+
+  useEffect(() => {
+    getUserFollowing();
+  }, [feedType]);
 
   useEffect(() => {
     getRecipes();
-  }, []);
+  }, [myUserFollowing]);
+
+  const getUserFollowing = async (userId) => {
+    try {
+      const response = await fetch(process.env.REACT_APP_API_URL + '/user/' + myUserName, {
+        method: 'GET',
+        headers: {
+          'Authorization': `Bearer ${token}`,
+          'Content-Type': 'application/json'
+        },
+      });
+      if (!response.ok) {
+        throw new Error('Failed to fetch user data');
+      }
+      const data = await response.json();
+      setMyUserFollowing(data.following || []);
+    } catch (error) {
+      console.error('Error fetching user data:', error);
+    }
+  };
+
+  const getRecipesFollowing = async () => {
+    setLoading(true);
+    const allFollowingRecipes = [];
+    for (const following of myUserFollowing) {
+      try {
+        const response = await fetch(process.env.REACT_APP_API_URL + '/recipe/user/' + following, {
+          method: 'GET',
+          headers: {
+            'Authorization': `Bearer ${token}`,
+            'Content-Type': 'application/json'
+          },
+        });
+        if (!response.ok) {
+          throw new Error('Failed to fetch recipes for following');
+        }
+        const followingRecipes = await response.json();
+        allFollowingRecipes.push(...followingRecipes.slice(0, maxRecipesPerUser));
+        console.error(followingRecipes)
+      } catch (error) {
+        console.error('Error fetching recipes for following:', error);
+      }
+    }
+  
+    setRecipes(allFollowingRecipes);
+    setLoading(false);
+  };
 
   const getRecipes = () => {
-    fetch(process.env.REACT_APP_API_URL + "/recipe/")
-      .then((response) => response.json())
-      .then((data) => {
-        console.log(data)
-        setRecipes(data);
-      })
-      .catch((error) => console.error("Error al obtener recetas:", error));
-  }
+    setLoading(true);
+    if (feedType === "following") {
+      getRecipesFollowing();
+    } else {
+      let url = process.env.REACT_APP_API_URL + "/recipe/";
+      fetch(url)
+        .then((response) => response.json())
+        .then((allRecipes) => {
+          const filteredRecipes = allRecipes.filter(recipe => 
+            !myUserFollowing.includes(recipe.username)
+          );
+          setRecipes(filteredRecipes);
+          setLoading(false);
+        })
+        .catch((error) => { 
+          console.error("Error al obtener recetas:", error); 
+          setLoading(false); 
+        });
+    }
+};
```

Comment:
getUserFollowing is fine. I am not convinced by the fact that we use the same recipes list for all. I believe we should store each list separately. Just remove the if from getRecipes and modify the useEffects accordingly. Add a third if necessary.